### PR TITLE
ENG-120: Cost Attribution

### PR DESCRIPTION
**Pull request by Iamdavidonuh**
This adds the ability to annotate pipelines with tags and Clean ups previous Parquet implementation

### CODE REVIEW

File: src/tracer_client.rs [hunk 1]
```diff
@@ -519,4 +520,51 @@ mod tests {
 
         Ok(())
     }
+
+    #[tokio::test]
```
Comment (MicheleVerriello) [inline src/tracer_client.rs:524]: we should cover edge cases but for now is okay

**Review by MicheleVerriello — APPROVED** [review]
It's a good job, when the future we should add also tests for unexpected behaviour and logs in the fuctions

File: src/tracer_client.rs (hunk 1 above)
Comment (Iamdavidonuh) [reply src/tracer_client.rs:524]: What edge case do you mean?

**Comment by Iamdavidonuh** [comment]
> It's a good job, when the future we should add also tests for unexpected behaviour and logs in the fuctions

Yeah that's definitely in the plan, for now the test only checks that tags appear in the event struct
